### ISSUE TITLE
quick-mod tools at the very bottom of a topic doesn't function 100%

### ISSUE THREAD

**Issue by tas2580** [tas2580/phpBB-3-1-SEO-URLs#34]
https://www.phpbb.com/community/viewtopic.php?f=456&t=2288486&start=120#p14594016

**Comment by HanPrower** [comment]
@tas2580 Lock Topic, Delete Topic and Change to Sticky/Annoucement/Global still do not work for me on 2 forums.

**Comment by HanFox** [comment]
@tas2580 This is still an issue.

**Comment by HanPrower** [comment]
Okay, so I think I know why this happens.

The ajax is looking for ./mcp.php which obviously doesn't exist.

You can see the path it's trying to use if you edit {ROOT}/assets/javascript/core.js and around line 357 add `console.log(res.S_CONFIRM_ACTION);`

If you hard code that to use the correct path for mcp.php it will perform the action, but 404 the redirect.

Hopefully this helps you find a fix @tas2580.  I see where it gets that link from, but I'm not sure what populates the form action in the #phpbb_confirm box that pops up when you select a quick mod action.

**Comment by tas2580** [comment]
The URL is generated in includes/functions.php on 2214
`'S_CONFIRM_ACTION'	=> str_replace('&amp;', '&', $u_action), //inefficient, rewrite whole function`

The problem is that there is no event to overwrite it.

**Comment by HanPrower** [comment]
Ah, I see, so will you be [requesting the event be added](https://area51.phpbb.com/phpBB/viewtopic.php?f=126&t=47511)?

If so, can you post the issue link. Be nice to keep an eye on it. This mod is a bit unfriendly without that functionality.

**Comment by 1datatec** [comment]
So is there any way for us to fix it?

**Comment by HanPrower** [comment]
I've been fiddling, and I'm not sure this is a good fix, but it so far works for me (obviously this is a hard-coded solution, but it may be possible to get to work cleanly as phpBB does have a script path variable):

I modified the `append_sid` function:
```
	public function append_sid($event)
	{
		if (strpos($event['url'], '/community/') === FALSE) {
			$url = preg_replace('/\.+\//', '', $event['url']);
			$event['url'] = '/community/'.$url;
		}
```

`/community/` is my script path (i.e. `https://[domain]/community/`).

What this is meant to do is replace general relative paths with absolute paths relating to the script path, so, `./mcp.php` or `./../mcp.php` would be rewritten to `/community/mcp.php`.

This fix also resolves another issue that occurs when phpBB rewrites #unread urls to #postid urls using JavaScript.

I haven't noticed any issues yet, and other extensions seem to still work (such as Topic Solved).

I will forewarn I've made other edits to the extension, but I don't think they interfere with this (in fact this fix has meant I can revert some of my other changes anyway).

**Comment by tas2580** [comment]
Simple Fix:
`RewriteRule ^(.*)-f([0-9]*)/mcp.php(.*) mcp.php?%{QUERY_STRING} [L,R=301]`

**Comment by 1datatec** [comment]
Can you please provide solution for nginx too?

**Comment by tas2580** [comment]
Should be:
`rewrite ^/(.*)-f([0-9]*)/mcp.php(.*) mcp.php?$query_string;`

This should be the first rule.

**Comment by 1datatec** [comment]
Didnt help on nginx. Still getting Ajax not found.
Here is the link from Delete topic: https://combatgaming.club/mcp.php?f=10&t=435&start=0&quickmod=1&redirect=.%2Fviewtopic.php%3Ff%3D10%26t%3D435&action=delete_topic&sid=7eb49745278f999fd5ff44aa0ab1fbac
Here is how topic link looks like: https://combatgaming.club/suggestions-f10/test-t435.html
Here are my rewrite rules:
```
	location / {
		rewrite ^/(.*)-f([0-9]*)/mcp.php(.*) mcp.php?$query_string;
		rewrite ^/(.*)-f([0-9]*)/(.*)-t([0-9]*)-s([0-9]*).html /viewtopic.php?f=$2&t=$4&start=$5&$query_string last;
		rewrite ^/(.*)-f([0-9]*)/(.*)-t([0-9]*).html /viewtopic.php?f=$2&t=$4&$query_string last;
		rewrite ^/(.*)-f([0-9]*)/index-s([0-9]*).html /viewforum.php?f=$2&start=$3&$query_string last;
		rewrite ^/(.*)-f([0-9]*)/ /viewforum.php?f=$2&$query_string last;
		rewrite ^/(.*)-f([0-9]*) /viewforum.php?f=$2&$query_string last;
		try_files $uri $uri/ /app.php?a=$uri;
	}
```
It does work if i open the Delete link in new window.

**Comment by Mightymax44** [comment]
Hi
the fix not work in my phpbb 3.2.3 with the last master branch.
I use apache.
